### PR TITLE
Add minimal CI

### PR DESCRIPTION
**Pull request by strangesource**
## Problem
Failing builds are not reported on pull requests

## Change
- Introduce minimal GitHub Actions CI workflow

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: '17'
+          architecture: ${{ inputs.architecture }}
+      - run: ./gradlew assemble
```
Comment (matamegger) [inline .github/workflows/ci.yaml:21]: Would it make sense to use `build`. It assembles and also dose some further verification and checks.
```suggestion
      - run: ./gradlew build
```

Comment (strangesource) [reply .github/workflows/ci.yaml:21]: Sounds good. 👍

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: '17'
+          architecture: ${{ inputs.architecture }}
```
Comment (matamegger) [inline .github/workflows/ci.yaml:20]:
```suggestion
```

Just noticed, there is no input for the architecture.

**Review by matamegger — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (strangesource) [reply .github/workflows/ci.yaml:20]: AII

**Pull request merged**